openedx/core/djangoapps/catalog/management/commands/cache_course_runs.py

```diff
@@ -0,0 +1,147 @@
+    updating these cache entries.
+    """
+    help = "Rebuild the LMS' cache of course runs data."
+
+    # lint-amnesty, pylint: disable=bad-option-value, unicode-format-string
+    def handle(self, *args, **options):  # lint-amnesty, pylint: disable=too-many-statements
+        failure = False
+
+        catalog_integration = CatalogIntegration.current()
+        username = catalog_integration.service_username
+
+        try:
+            user = User.objects.get(username=username)
+        except User.DoesNotExist:
+            logger.exception(
+                f'Failed to create API client. Service user {username} does not exist.'
+            )
+            raise
+
+        all_courserun_keys = CourseOverview.get_all_courses().values_list('id', flat=True)
+        for courserun_keys in self.chunks(all_courserun_keys):
+            client = get_catalog_api_client(user)
+            api_base_url = get_catalog_api_base_url()
+
+            course_details = self.fetch_courses_details(client, courserun_keys, api_base_url)
+            processed_courses_details = self.process_course_details(courserun_keys, course_details)
+
+            logger.info(f'Caching details for {courserun_keys} courses.')
+            # TODO: Cache course details by course run key
+            cache.set_many(processed_courses_details, None)
+
+        if failure:
+            sys.exit(1)
+
+    @classmethod
+    def chunks(cls, keys, chunk_size=50):
+        """
+        Yield chunks of size `chunk_size`
+        """
+        for i in range(0, len(keys), chunk_size):
+            yield keys[i:i + chunk_size]
+
+    @classmethod
+    def fetch_courses_details(cls, client, courserun_keys, api_base_url):  # lint-amnesty, pylint: disable=missing-function-docstring
+        """
+        Fetch the course data from discovery using /api/v1/courses endpoint
+        """
+        course_keys = [cls.construct_course_key(courserun_key) for courserun_key in courserun_keys]
+        encoded_course_keys = ','.join(map(quote_plus, course_keys))
+
+        logger.info(f'Fetching details for courses {course_keys}.')
+        api_url = f"{api_base_url}/courses"
+        response = client.get(api_url, params={"keys": encoded_course_keys})
+        response.raise_for_status()
+        courses_details = response.json()
+
+        return courses_details
+
+    @classmethod
+    def process_courses_details(cls, courserun_keys, courses_details):
+        """
+        Parse and extract the minimal data that we need.
+        """
+        courses = {}
+        for courserun_key in courserun_keys:
+            course_metadata = cls.find_attr(courses_details, 'key', courserun_key)
+
+            course_type = course_metadata.get('course_type')
+            product_source = course_metadata.get('product_source')
+            if product_source:
+                product_source = product_source.get('slug')
+
+            if course_type == EXEC_ED_COURSE_TYPE:
```

Comment:
Can we also inject the COURSE_HOST_URL = "https://www.getsmarter.com" here ... this would simplify the architecture for https://github.com/openedx/edx-platform/pull/32275#discussion_r1199817608